### PR TITLE
Add docs for library organization best practices

### PR DESCRIPTION
**Pull request by nre-ableton**
I'm improving the documentation in the `README` to add some best-practices regarding library organization. This is based mostly on a [recent discussion](https://github.com/jenkinsci/JenkinsPipelineUnit/issues/512#issuecomment-1096851764) in another issue, but also because I find myself copy-pasting the same advice to several other issues.

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by nre-ableton** [comment]
Since I'm pretty much the only active maintainer of this repo these days, I'll merge this PR in one week unless there are any objections. Reviews would be appreciated, though.

File: README.md
Comment (UlrichEckhardt) [inline README.md:825]: "...complexity to external scripts..." -- missing "to"

File: README.md
Comment (UlrichEckhardt) [inline README.md:838]: I know that CodeNarc can be used on regular Groovy code and Jenkinsfiles. Is that worth mentioning here?

**Review by UlrichEckhardt — APPROVED** [review]
Looks like a good step forward!

File: README.md
Comment (nre-ableton) [reply README.md:838]: It is (and I do love that tool!), but it's nothing compared to the linters available in other languages. Still, I will mention it, though.

File: README.md
Comment (stchar) [inline README.md:910]: `new HardMath(..)`  is awesome. I used to create pipeline script just as `EmptyPipeline` but to load the library and to return an instance of my library class.  But your approach is way way way better.

**Review by stchar — APPROVED** [review]
looks good to me

**Pull request merged**
